0018-connector-risk-mitigations/0018-connector-risk-mitigations.md

```diff
@@ -0,0 +1,40 @@
+* **Adjusting Transfer Expiry Window** - connectors determine the window of time they require between the incoming and outgoing transfer expires and they should set this window such that they are confident that they will be able to deliver the fulfillment in time even if the outgoing transfer is executed at the last possible moment
+* **Denial of Service Protections** - connectors should take standard [Denial of Service](https://en.bitcoin.it/wiki/Weaknesses#Denial_of_Service_.28DoS.29_attacks) attack prevention mechanisms to prevent attackers from overloading the connectors' servers with network traffic
+* **Prioritizing Fulfillments** - connectors should prioritize fulfilling transfers over preparing new ones or responding to quote requests and may even have separate processes or machines responsible for those different behaviors
+* **Redundant Instances** - connectors should run redundant processes and machines to increase the difficulty of interfering with their processing
+* **Avoiding Public Endpoints** - connectors may reduce attack vectors by using communication channels and protocols that do not entail having a publicly-accessible server (for example using [WebRTC](https://webrtc.org/) or [Virtual Private Networks](https://en.wikipedia.org/wiki/Virtual_private_network) to communicate with peers
+
+## Liquidity Exhaustion
+
+Attackers could tie up connectors' liquidity by preparing payments through them that the attacker knows will fail, for example if the attacker sends themselves numerous payments and then does not fulfill any of them.
+
+Possible mitigations include:
+
+* **Managing "Payment Bandwidth"** - connectors should monitor the percentage of their total liquidity that is on hold at any given time for a given peer or customer and may reject incoming payment requests if that party exceeds their allocated "payment bandwidth". Connectors may allocate less bandwidth to unknown or untrustworthy senders. Note that ILP Packets do not contain the source account so connectors should apply this logic to the immediate peer or customer that they receive the payments from
+* **Preferring Smaller Payments** - smaller payments place a smaller percentage of the connector's liquidity on hold and thus each one carries less risk that the payment being prepared and then failing would create significant opportunity cost
+* **Blacklisting Senders and Receivers** - connectors may refuse to facilitate payments from certain sources or to certain destinations if their rate of failed payments is unusually high
+
+## Exchange Rate Volatility
+
+Once connectors prepare their outgoing transfers, they are committed to the payment even if the exchange rate between the assets fluctuates.
+
+Possible approaches include:
+
+* **Building a Buffer into the Rate** - connectors may add a buffer to their expected exchange rate to account for some movement in the price, and they may add a premium for especially volatile currencies
```

Comment:
Rather than "building a buffer into the rate", the term of art is probably more like, "Accounting for Slippage"